### DISCUSSION TITLE
Backstage - Jenkins Templating Engine Integration

### DISCUSSION THREAD

**Discussion by regicsolutions** [jenkinsci/templating-engine-plugin#164] [Ideas]
### Feature Request

[backstage](https://backstage.io/docs/features/software-templates/software-templates-index) is an opensource platform for building developer portals and provides the front end/orchestration layer where approved templates are housed. The platform is extremely extensible and allows you to create your own templaters,  it makes use of cookiecutter as an out of the box templating engine, would be interesting to see integration with the Jenkins Templating engine where a developer would be able to select from different approved pipeline templates! 

Heres a quick video that gives you a quick overview of the Software Templates feature: https://www.youtube.com/watch?v=U1iwe3L5pzc 

https://github.com/backstage/backstage/blob/c09063ae815333c6616d8e7564403d33e7f6326f/microsite/blog/2020-08-05-announcing-backstage-software-templates.md

**Comment by steven-terrana** [comment]
Definitely an interesting idea! I'm not sure this integration would come directly as part of the JTE core plugin but perhaps another jte-backstage plugin could be written that implements the appropriate extension points to let backstage serve as a JTE governance tier.

**Comment by regicsolutions** [comment]
> Definitely an interesting idea! I'm not sure this integration would come directly as part of the JTE core plugin but perhaps another jte-backstage plugin could be written that implements the appropriate extension points to let backstage serve as a JTE governance tier. 

Yep, exactly what I was thinking
